### ISSUE TITLE
Execute Job don't clean subproccess jobs after finishes

### ISSUE THREAD

**Issue by mliradelc** [opencast/opencast#2034]
**Describe the bug**
When using the execute WOH and the workflow is set to retrieve a "Track" object,  Opencast creates a Job for inspecting the new created track, when the inspection job finishes, is left an empty job in "waiting" state.

**Example**:
Here is the log with Debug in TRACE mode for the Job barrier and the execute module:
```bash
020-11-02 16:17:23,179 | DEBUG | (ExecuteOnceWorkflowOperationHandler:117) - Running execute workflow operation with ID 4332
2020-11-02 16:17:23,181 | DEBUG | (JobBarrier:161) - No waiting job set, unable to put waiting job into waiting state
2020-11-02 16:17:27,345 | DEBUG | (AbstractJobProducer:176) - 639 Current load on this host: 1.0, job's load: 1.0, job's status: DISPATCHING, max load: 12.0
2020-11-02 16:17:27,346 | DEBUG | (AbstractJobProducer:199) - 639 Accepting job 4333 of type org.opencastproject.execute with load 1 because load of 2 is within this node's limit of 12.
2020-11-02 16:17:27,920 | INFO  | (ExecuteServiceImpl:494) - Running command /home/max/repos/mldc-opencast/build/opencast-dist-develop-8-SNAPSHOT/etc/overlay_publish.sh
2020-11-02 16:17:27,920 | DEBUG | (ExecuteServiceImpl:495) - Starting subprocess /home/max/repos/mldc-opencast/build/opencast-dist-develop-8-SNAPSHOT/etc/overlay_publish.sh with arguments /media/max/40423f62-2aad-4b51-b563-ac55b41b4a241/repos/mldc-opencast/build/opencast-dist-develop-8-SNAPSHOT/data/opencast/workspace/http_localhost_8080/assets/assets/488aca11-3a03-496d-97f2-1f08d72899d3/1f2d02be-f1f7-42e1-b55a-e68878023aee/1/dual_presentation_1.mp4, /media/max/40423f62-2aad-4b51-b563-ac55b41b4a241/repos/mldc-opencast/build/opencast-dist-develop-8-SNAPSHOT/data/opencast/workspace/http_localhost_8080/assets/assets/488aca11-3a03-496d-97f2-1f08d72899d3/21e23cf7-6b17-4b41-8eac-c004a5cf57bc/1/dual_presenter_1.mp4, /media/max/40423f62-2aad-4b51-b563-ac55b41b4a241/repos/mldc-opencast/build/opencast-dist-develop-8-SNAPSHOT/data/opencast/workspace/http_localhost_8080/assets/assets/488aca11-3a03-496d-97f2-1f08d72899d3/21e23cf7-6b17-4b41-8eac-c004a5cf57bc/1/4333_composite_pip.mp4
2020-11-02 16:17:28,182 | TRACE | (JobBarrier$JobStatusUpdater:313) - Job {id:4333, operation:Execute_Mediapackage, status:QUEUED} is still in the works
2020-11-02 16:17:33,183 | TRACE | (JobBarrier$JobStatusUpdater:313) - Job {id:4333, operation:Execute_Mediapackage, status:QUEUED} is still in the works
2020-11-02 16:17:38,184 | TRACE | (JobBarrier$JobStatusUpdater:313) - Job {id:4333, operation:Execute_Mediapackage, status:QUEUED} is still in the works
2020-11-02 16:17:43,185 | TRACE | (JobBarrier$JobStatusUpdater:313) - Job {id:4333, operation:Execute_Mediapackage, status:QUEUED} is still in the works
2020-11-02 16:17:48,186 | TRACE | (JobBarrier$JobStatusUpdater:313) - Job {id:4333, operation:Execute_Mediapackage, status:QUEUED} is still in the works
2020-11-02 16:17:53,187 | TRACE | (JobBarrier$JobStatusUpdater:313) - Job {id:4333, operation:Execute_Mediapackage, status:QUEUED} is still in the works
2020-11-02 16:17:58,188 | TRACE | (JobBarrier$JobStatusUpdater:313) - Job {id:4333, operation:Execute_Mediapackage, status:QUEUED} is still in the works
2020-11-02 16:18:00,410 | DEBUG | (ExecuteServiceImpl:504) - Command /home/max/repos/mldc-opencast/build/opencast-dist-develop-8-SNAPSHOT/etc/overlay_publish.sh finished with result 0
2020-11-02 16:18:01,793 | DEBUG | (ExecuteServiceImpl:512) - Deleted the local copy of the encoded file at /media/max/40423f62-2aad-4b51-b563-ac55b41b4a241/repos/mldc-opencast/build/opencast-dist-develop-8-SNAPSHOT/data/opencast/workspace/http_localhost_8080/assets/assets/488aca11-3a03-496d-97f2-1f08d72899d3/21e23cf7-6b17-4b41-8eac-c004a5cf57bc/1/4333_composite_pip.mp4
2020-11-02 16:18:03,193 | DEBUG | (JobBarrier:178) - No waiting job set, unable to put waiting job into waiting state
2020-11-02 16:18:03,199 | DEBUG | (JobBarrier:153) - Job 4333 set to WAITING state.
2020-11-02 16:18:07,387 | DEBUG | (AbstractJobProducer:176) - 650 Current load on this host: 0.0, job's load: 0.2, job's status: DISPATCHING, max load: 12.0
2020-11-02 16:18:07,388 | DEBUG | (AbstractJobProducer:199) - 650 Accepting job 4334 of type org.opencastproject.inspection with load 0,2 because load of 0,2 is within this node's limit of 12.
2020-11-02 16:18:08,232 | DEBUG | (JobBarrier:170) - Job 4333 wakened and set back to RUNNING state.
2020-11-02 16:18:08,241 | DEBUG | (ExecuteOnceWorkflowOperationHandler:230) - Execute operation 4332 completed
```

Job 4332 is created for Execute WOH, the script that is inside, starts FFMPEG, this creates a subprocess, opencast gaves to this the Job 4333. The FFMPEG finishes and then the script also finishes. Opencast finishes the Job 4332 but not the Job 4333, leaving the Job empty running.

**To Reproduce**
1. Create a workflow that includes "execute-once" WOH executing a process that generates a Track object.
2. Run the workflow, wait until completes
3. See in the servers -> Jobs section there is a process still running.

**Environment Information**
 - OS: Ubuntu 18.10
- Opencast 8.7

Edit: Found the exact cause of the bug

**Comment by mliradelc** [comment]
Video Showing the bug:
https://www.youtube.com/watch?v=L3l-TsWoyFM

Also, I found the code that creates the issue:
`modules/execute-workflowoperation/src/main/java/org/opencastproject/execute/operation/handler/ExecuteOnceWorkflowOperationHandler.java`
```java
if (resultElement.getElementType() == MediaPackageElement.Type.Track) {
            // Have the track inspected and return the result
            Job inspectionJob = null;
            inspectionJob = inspectionService.inspect(resultElement.getURI());
            JobBarrier barrier = new JobBarrier(job, serviceRegistry, inspectionJob);
            if (!barrier.waitForJobs().isSuccess()) {
              throw new ExecuteException("Media inspection of " + resultElement.getURI() + " failed");
            }

            resultElement = MediaPackageElementParser.getFromXml(inspectionJob.getPayload());
}
```

**Comment by karendolan** [comment]
Closed with commit 4959712 in r/8.x
